content/blog/2022/01/mutating-policy-behave-as-validating.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Kubewarden has policies to cover all the Kubernetes Pod Security Policies.
+authors:
+	- José Guilherme Vanz
+date: 2022-01-31
+---
+
+We at the Kubewarden team has been doing an effort to allow our users to have
+policies equivalent for all the deprecated Pod Security Policies (PSP).
+So, we wrote policies to replicate the same validations available in the Kubernetes
+PSPs. During this work, we counted on the community help to map and validate
+the policies.
+
+Since Kubewarden controller <version-placeholder> and Policy Server version
+<version-placeholder>, Kubewarden users have policies which replace all the PSPs!
+All the them can be found in the Policy Hub looking for the keyword "PSP".
+
+You may are thinking: why do I need to a specific versions to have all PSP policies?
+
+Actually, depending on what you would like to do, you do not need that. Let me
+explain.
+
+Before the version mentioned previously in this text. If a policy mutate a
+request and its "mutating" configuration field is 'true', the policy server
+always accept the request.  However, this is not in-line with PSP. Users
+should be able to reject a request if a mutation happen. To solve this issue,
+we change the Kubewarden controller and Policy Server to allow the user
+configure a mutating policy to behave like a validating one. If you do not need
+that, you can use the policies in previous Kubewarden versions.
```

Comment:
```suggestion
The answer is related to "requests mutation". Let me explain.

In the previous versions of Kubewarden Policy Server, if a policy mutated a request by setting the configuration field `mutating` to 'true', the policy server would always accept the request and this behavior is not aligned with the Kubernetes PSP validation. 

Instead, you should be able to reject a request even if a mutation happens. To solve this misalignment, the team changed the Kubewarden controller and Policy Server to allow you configure a mutating policy to behave like a validated one. And if you don't need it, you can use the policies in previous Kubewarden versions.
```

Reply:
I adapted the sentences, however review it as the meaning might have changed from what you wanted.

Reply:
I think that meaning is not quite right. The policy does not mutate the "mutating" field. The mutating field is a policy configuration . What I was trying to say is that if the policy **is** configured as "mutating" **and** this policy mutate a request (does not matter what field it mutate), the request is accepted. Let me adapt your sentence for I what I would like to say.